### PR TITLE
fix: #1241: Fix destination validation in the ibc-out form

### LINKED ISSUE #1241: IBC-out addresses incorrectly validating
It appears the boolean is opposite? 

Should work:
Screenshot 2024-06-03 at 12 45 48 PM

Should not work:
Screenshot 2024-06-03 at 12 45 54 PM

### PR DESCRIPTION
**Pull request by VanishMax**
Closes #1241 

I noticed the tests were skipped in the `ibc-out.test.ts` file. I un-skipped the passing tests and added a comment for the future to fix the broken tests

### CODE REVIEW

File: apps/minifront/src/state/ibc-out.ts [hunk 1]
```diff
@@ -199,15 +199,16 @@ export const ibcValidationErrors = (state: AllSlices) => {
...
+      : validateUnknownAddress(state.ibcOut.chain, state.ibcOut.destinationChainAddress),
     amountErr: !state.ibcOut.selection
       ? false
       : amountMoreThanBalance(state.ibcOut.selection, state.ibcOut.amount),
   };
 };
 
 /**
- * we don't know what format foreign addresses are in. so this only checks:
+ * Matches the given address to the chain's address prefix. Returns `true` if the address is not valid.
```
Comment (grod220) [inline apps/minifront/src/state/ibc-out.ts:210]: Hmm, think there could be a bug in `validateUnknownAddress`. It should return `true` if the address _is_ valid. That is consistent with the first line:

```
if (!chain || address === '') return false; // not valid = false
```

We should rename this function to: `unknownAddrIsValid()` to add clarity.

Comment (VanishMax) [reply apps/minifront/src/state/ibc-out.ts]: Right, updated!

File: apps/minifront/src/state/ibc-out.test.ts [hunk 2]
```diff
@@ -85,19 +86,30 @@ describe.skip('IBC Slice', () => {
...
+    test('destination address validation per selected chain', () => {
+      const osmoAddress = 'osmo1dyrr4r42ql4em7d46srcmnn5ymxk9asvcv95sg';
+
+      useStore.getState().ibcOut.setChain(chain);
+      useStore.getState().ibcOut.setDestinationChainAddress(osmoAddress);
+
+      const validationErrors = ibcValidationErrors(useStore.getState());
+
+      expect(validationErrors.recipientErr).toBeFalsy();
+    });
```
Comment (grod220) [inline apps/minifront/src/state/ibc-out.test.ts:112]: Can you add a test that triggers the validation error? Aka

```ts
   test('invalid address', () => {
      const osmoAddress = 'osmo1xxxxx';

      useStore.getState().ibcOut.setChain(chain);
      useStore.getState().ibcOut.setDestinationChainAddress(osmoAddress);

      const validationErrors = ibcValidationErrors(useStore.getState());

      expect(validationErrors.recipientErr).toBeTruthy();
    });
```

File: apps/minifront/src/state/ibc-out.ts [hunk 3]
```diff
@@ -199,23 +199,24 @@ export const ibcValidationErrors = (state: AllSlices) => {
...
     amountErr: !state.ibcOut.selection
       ? false
       : amountMoreThanBalance(state.ibcOut.selection, state.ibcOut.amount),
   };
 };
 
 /**
- * we don't know what format foreign addresses are in. so this only checks:
+ * Matches the given address to the chain's address prefix.
+ * We don't know what format foreign addresses are in. so this only checks:
```
Comment (grod220) [inline apps/minifront/src/state/ibc-out.ts:211]:
```diff
- * We don't know what format foreign addresses are in. so this only checks:
+ * We don't know what format foreign addresses are in, so this only checks:
```

**Review by grod220 — APPROVED** [review]
Good job! A few more comments and then good to merge.

**Pull request merged**